### PR TITLE
Ensure action cgroups are placed under the k8s container cgroup

### PR DESCRIPTION
**Pull request by bduffany**
Fixes https://github.com/buildbuddy-io/buildbuddy-internal/issues/4140

### CODE REVIEW

File: enterprise/server/remote_execution/cgroup/cgroup.go [hunk 1]
```diff
@@ -31,6 +32,153 @@ const (
...
+	if controllers := parts[1]; controllers != "" {
+		return "", ErrV1NotSupported
+	}
+	return strings.Join(parts[2:], ":"), nil
+}
+
+// Setup configures the cgroup at the given path with the given settings.
+// Any settings for cgroup controllers that aren't enabled are ignored.
+// IO limits are applied to the given block device, if specified.
+func Setup(ctx context.Context, path string, s *scpb.CgroupSettings, blockDevice *block_io.Device) error {
```
Comment (vanja-p) [inline enterprise/server/remote_execution/cgroup/cgroup.go:64]: i think this diff is from your previous PR

Comment (bduffany) [reply enterprise/server/remote_execution/cgroup/cgroup.go]: woops, fixed

File: enterprise/server/cmd/executor/executor_linux.go [hunk 2]
```diff
@@ -4,14 +4,43 @@ package main
...
+		if errors.Is(err, cgroup.ErrV1NotSupported) {
+			log.Warningf("Note: executor is running under cgroup v1, which has limited support. Some functionality may not work as expected.")
+			return "", nil
+		}
+	}
+	// Return the executor cgroup directly, for now.
+	// TODO(https://github.com/buildbuddy-io/buildbuddy-internal/issues/4110):
+	// restructure cgroups so that all actions are run in a single cgroup and
+	// the executor runs as a sibling of that.
+	return strings.TrimPrefix(relpath, string(os.PathSeparator)), nil
```
Comment (vanja-p) [inline enterprise/server/cmd/executor/executor_linux.go:41]: why not remove the `/` in cgroup.GetCurrent? That would make it easier to use.

**Review by vanja-p — APPROVED** [review]

File: enterprise/server/cmd/executor/executor_linux.go (hunk 2 above)
Comment (bduffany) [reply enterprise/server/cmd/executor/executor_linux.go]: no good reason - done

**Review by vadimberezniker — APPROVED** [review]
Fine with me as long as it doesn't interfere with us assigning CPUs to the executor cgroup, whether it's through k8s or through us messing with the cgroup directly.

**Comment by bduffany** [comment]
> Fine with me as long as it doesn't interfere with us assigning CPUs to the executor cgroup, whether it's through k8s or through us messing with the cgroup directly.

Discussed offline - this PR will make it so we can no longer control the executor cgroup independently from the action cgroups, but I will send a follow-up PR to properly structure the cgroups so that actions run in a sibling cgroup of the executor.

**Review by tylerwilliams — APPROVED** [review]

**Pull request merged**